content/posts/2025-01-03-packets-xlation-enrichment.md

```diff
@@ -0,0 +1,178 @@
+By attaching eBPF programs to network events, you can capture and enrich data about service flows.
+
+Here’s a high-level approach:
+
+Attach eBPF Programs:
+Use eBPF programs to hook into kernel networking events, such as kprobe on functions like `nf_nat_manip_pkt`.
+
+Enrich Flow Logs:
+As network packets are processed, augment flow logs with metadata about the translated endpoint. This can include:
+
+- Source Pod IP/Name
+
+- Source Port
+
+- Destination Pod IP/Name
+
+- Destination Port
+
+- Conntrack Zone ID
+
+- ICMP code
+
+Benefits of Using eBPF
+
+Granular Observability: Directly see which pod served a request.
+
+Low Overhead: Operates in the kernel with minimal impact on performance.
+
+Dynamic Updates: Respond to changes in Kubernetes without requiring application restarts.
+
+Simplified Architecture: No need for sidecars or intrusive network plugins.
+
+## How to enable Packet Translation enrichment feature
+
+ To enable packet translation enrichment feature, need to create a flow collector with the following feature enabled
+
+```yaml
+apiVersion: flows.netobserv.io/v1beta2
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  agent:
+    type: eBPF
+    ebpf:
+      features:
+      - "PacketTranslation"
+```
+
+## Example
+
+ To demonstrate this feature, let's configure a `CluesterIP` kubernetes service using the following example yaml
+
+```yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+ name: hello-pod
+ namespace: xlat-test
+ labels:
+  app: hello-pod
+spec:
+ containers:
+ - name: hello-world
+   image: gcr.io/google-samples/node-hello:1.0
+   ports:
+   - containerPort: 8080
+     protocol: TCP
+---
+kind: Service
+apiVersion: v1
+metadata:
+ name: hello-pod
+ namespace: xlat-test
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: hello-pod
+  type: ClusterIP
+```
+
+check the created service to find the `CluterIP` and `Port`
+
+```bash
+$oc describe svc -n xlat-test
+Name:              hello-pod
+Namespace:         xlat-test
+Labels:            <none>
+Annotations:       <none>
+Selector:          app=hello-pod
+Type:              ClusterIP
+IP Family Policy:  SingleStack
+IP Families:       IPv4
+IP:                172.30.165.151
+IPs:               172.30.165.151
+Port:              http  80/TCP
+TargetPort:        8080/TCP
+Endpoints:         10.129.0.37:8080
+Session Affinity:  None
+Events:            <none>
+
+$oc get pods -n xlat-test
+oc get pods -n xlat-test
+NAME        READY   STATUS    RESTARTS   AGE
+client      1/1     Running   0          5s
+hello-pod   1/1     Running   0          8m54s
+
+```
+
+Then we can send traffic to this service IP and check the enriched flows on the netobserv console
+
+```bash
+$ while true; do oc exec -i -n xlat-test client -- curl 172.30.165.151:80 ; sleep 1; done
+```
+
+from Netobserv console let's filter on Traffic destination `Kind` is `Service` in `xlat-test` namespace
```

Comment:
```suggestion
4. From the network observability console **Network Traffic** page, click the *Taffic flows* tab  and filter on Traffic destination `Kind` is `Service` in the `xlat-test` namespace:
```

Reply:
I suggest clarifying the exact area of the console where they'll need to be working.